isisd/isis_mt.c

```diff
@@ -398,7 +398,7 @@ bool tlvs_to_adj_mt_set(struct isis_tlvs *tlvs, bool v4_usable, bool v6_usable,
 		    && !tlvs->mt_router_info_empty) {
 			/* Other end does not have MT enabled */
 			if (mt_settings[i]->mtid == ISIS_MT_IPV4_UNICAST
-			    && v4_usable)
+			    && (v4_usable || v6_usable))
```

Comment:
I would have expected something like:

`(mt_settings[i]->mtid == ISIS_MT_IPV4_UNICAST && v4_usable) || (mt_settings[i]->mtid == ISIS_MT_IPV6_UNICAST && v6_usable)`

But speaking to @manuhalo he told me `ISIS_MT_IPV6_UNICAST` doesn't exist; maybe it would be a good idea to rename it to `ISIS_MT_IP_UNICAST` or something alike

Reply:
My apologies, I was not very clear: it does exist but it is only used when MT is enabled. `ISIS_MT_IPV4_UNICAST` is the default topology when MT is disabled, but also doubles up as, well, IPv4 unicast when mt is enabled. Ideally the default non-MT topology would be separate from either the IPv4 or the IPv6 unicast topologies, but that is a more complicated issue to solve

Reply:
As far as I am aware, there is no IPv4 unicast topology other than topo 0 in IS-IS.

Generally, I would recommend to always route IPv6 using a separate topology and leave topo 0 for IPv4. Otherwise, you will have interesting failure modes if you have any link in your topology which only supports one of the two address families, as this fact cannot be flooded.

Maybe @riw777 has a recommendation here?

Reply:
@cfra -- correct on all counts.